pkg/operator/injection/injection.go

```diff
@@ -68,15 +54,21 @@ func GetControllerName(ctx context.Context) string {
 // dynamically at runtime due to the necessity of having to build logic around re-queueing to ensure that settings are
 // properly reloaded for things like feature gates
 func WithSettingsOrDie(ctx context.Context, kubernetesInterface kubernetes.Interface, settings ...settings.Injectable) context.Context {
-	cancelCtx, cancel := context.WithCancel(ctx)
+	cancelCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
```

Comment:
```suggestion
	cancelCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
```

I did 5 seconds in my other PR and that should be more than enough time for the ConfigMap to be added and the change to be seen